tests/i18n/tests.py

```diff
@@ -1927,15 +1927,15 @@ def test_other_lang_with_prefix(self):
         self.assertEqual(response.content, b"Oui")
 
     def test_unprefixed_language_with_accept_language(self):
-        """'Accept-Language' is respected."""
+        """'Accept-Language' is ignored."""
         response = self.client.get("/simple/", headers={"accept-language": "fr"})
-        self.assertRedirects(response, "/fr/simple/")
+        self.assertEqual(response.status_code, 200)
 
     def test_unprefixed_language_with_cookie_language(self):
-        """A language set in the cookies is respected."""
+        """A language set in the cookies is ignored."""
         self.client.cookies.load({settings.LANGUAGE_COOKIE_NAME: "fr"})
         response = self.client.get("/simple/")
-        self.assertRedirects(response, "/fr/simple/")
+        self.assertEqual(response.status_code, 200)
```

Comment:
@felixxm @shangxiao I needed to update these tests to comply with the new suggestion.
What is the expected behaviour in this case, am I right to update this to respect the path or should it redirect?

I think @aboutofpluto is suggesting it should respect the path in this case.

(On the new suggestion it ideally would need a depreciation note so it's not the most ideal "solution", but I'm still using this pr to think out loud)

Reply:
Right looking at the commit which added the behaviour, the redirect was kinda the whole point :sweat_smile: 

Which would mean that I think [the original attempt](https://github.com/django/django/commit/566fbe353ce9fab374373e9cf7b1404d2d57f39d) "fixed" the bug with translations and the cookie (and I could probably add a tweak for the header case).
So I am going to continue with that, otherwise the discussion is more whether the commit should be reverted.